File: Java scripts/Graph Algorithms/Dfs.java
import java.io.*;
import java.util.*;
class Graph 
{
  private int V;
  private LinkedList<Integer> adj[];
	Graph(int v)
  {
    V = v;
    adj = new LinkedList[v];
    for (int i = 0; i < v; ++i) adj[i] = new LinkedList();
  }
  void addEdge(int v, int w)
  {
  	adj[v].add(w);
		adj[w].add(v);
  }
  void DFSUtil(int v, boolean visited[], String s)
  {
    visited[v] = true;
    s += v+" ";
    Iterator<Integer> i = adj[v].listIterator();
    while (i.hasNext()) 
		{
      int n = i.next();
      if (!visited[n])
        DFSUtil(n, visited);
    }
		return s.trim();
  }
  void DFS(int v)
  {
    boolean visited[] = new boolean[V];
    DFSUtil(v, visited);
  }
}

File: cpp/graph_algorithms/dfs.cpp
#include <bits/stdc++.h>
using namespace std;
 
class Graph {
public:
    map<int, bool> visited;
    map<int, list<int> > adj;
    void addEdge(int v, int w);
    void DFS(int v);
};
 
void Graph::addEdge(int v, int w)
{
  adj[v].push_back(w);
}
 
void Graph::DFS(int v, string s)
{
  visited[v] = true;
  s += v+" ";
  for (list<int>::iterator i = adj[v].begin(); i != adj[v].end(); ++i)
    if (!visited[*i])
      DFS(*i);
	return s;
}